app/src/main/java/com/sdp13epfl2021/projmag/activities/QRCodeActivity.kt

```diff
@@ -0,0 +1,20 @@
+package com.sdp13epfl2021.projmag.activities
+
+import android.graphics.BitmapFactory
+import android.os.Bundle
+import android.widget.ImageView
+import androidx.appcompat.app.AppCompatActivity
+import com.sdp13epfl2021.projmag.R
+
+
+class QRCodeActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_q_r_code)
+
+        val imageView = findViewById<ImageView>(R.id.qrcode_image)
+        val byteArray = intent.getByteArrayExtra("qrcode")
+        val bmp = BitmapFactory.decodeByteArray(byteArray, 0, byteArray!!.size)
```

Comment:
the `byteArray!!.size` throw a NullPointerException which cause the app to crash in case of issue.
Maybe consider using :
```kotlin
byteArray?.let{ bytes ->
    BitmapFactory....
    imageView....
}
```

Reply:
I agree, and will add this improvement